src/lib/support/c/CHIPPlatformMemory.cpp

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2020 Project CHIP Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ *    @file
+ *          Provides a C-callable wrapper around CHIPMem.h.
+ */
+
+#include <support/CHIPMem.h>
+#include <support/c/CHIPPlatformMemory.h>
+
+extern "C" {
+
+extern int CHIPPlatformMemoryInit(void * buf, size_t bufSize)
```

Comment:
Why can't this just be in CHIPMem.h and the rest hidden by ifdef __cplusplus ?

Reply:
Do you mean the declarations, or are you suggesting defining them inline in the header?

Reply:
I meant the declarations. Not sure if it's better, perhaps separate is the way to go.

Reply:
The rationale for making them separate is to avoid adding non-namespaced declarations to a common header.